codec-http2/src/main/java/io/netty/handler/codec/http2/Http2MultiplexHandler.java

```diff
@@ -111,7 +116,10 @@ public void operationComplete(ChannelFuture future) {
      *
      * @param inboundStreamHandler the {@link ChannelHandler} that will be added to the {@link ChannelPipeline} of
      *                             the {@link Channel}s created for new inbound streams.
+     * @deprecated Use {@link #forClient(ChannelHandler, ChannelHandler)} (with the second parameter as {@code null})
+     * or {@link #forServer(ChannelHandler)} instead.
      */
+    @Deprecated
```

Comment:
is deprecation actually necessary? This change helps for small edge case (testing, with EmbeddedChannel only - but many use real channels for testing), otherwise current API is sufficient. It would force netty users onto new API for no practical gain.

Reply:
it's not like they have to be removed in 4.x, but it seems weird to have both the factory and the constructor.